### DISCUSSION TITLE
Unable to create account. sl-db error?

### DISCUSSION THREAD

**Discussion by whisperdancer** [simple-login/app#1324] [Self-hosting question]
This is a dup because the previous discussion encountered spam/phising comments. Please delete #1321. Thank you.

I followed instructions exactly in Readme.md. I've had more trouble getting this service to work than any other service and I self host many. Super frustrating experience.

Create a new account and get a server error: Check logs and appears to be a problem with the db structure.

docker logs sl-db has a ton of errors, but this seems to be the important line:
2022-09-28 17:20:39.353 UTC [77] ERROR: column public_domain.can_use_subdomain does not exist at character 258

What is causing this? And how to fix it? Thank you.

**Comment by nguyenkims** [comment]
@whisperdancer I deleted the other discussion. Are you using the latest version which is  https://github.com/simple-login/app/tree/4.6.2-beta? The error might indicate the database migration isn't properly run. Please make sure to run all steps in the README as they are all needed to get SL running.

**Reply by whisperdancer** [reply]
Yes, I have gone through the readme each step one at a time. Only difference is I have converted the docker run to compose stack. 

From the compose stack, I do the following:
1. docker-compose start postgres
2. docker-compose start migration
3. docker.compose start init
Check logs for errors. None, logs clean.
4. docker-compose start to bring up remaining containers in proper order (depends_on is set in compose)
Check logs again, no errors. clean.

Navigate to https://simple.mydomain.com. Site comes up. Try and create a new user and server errors.

I started with 3.4.0, then tried the beta 4.6.2.-beta. As I thought it might be db corruption or the schema didn't get updated properly. I stopped all the containers, deleted all files under sl/db and started postgres again. Then I run the migration and init commands before bringing up the remaining containers. I've been getting all sorts of different errors when trying to create my first account. 

When using the latest 4.6.2-beta, should I first install 3.4.0, then run migration and init on db previously done wiht 3.4.0 or should I clean db files and run migration and init on a new clean db?

**Reply by nguyenkims** [reply]
If you don't have any data yet, I'd suggest to follow the readme and start with  4.6.2-beta right away (without installing previous versions). 
We don't use docker-compose and unfortunately never test using it with SL.

**Reply by whisperdancer** [reply]
I have no data because I'm unable to create an account. I have tried it strictly your way with no compose and with compose. Results are the same.

**Comment by whisperdancer** [comment]
I'm also using a reverse proxy and to make that work had to do this:

on the sl-app container, remove 127.0.0.1

Change: `-p 127.0.0.1:7777:7777 \ to -p 7777:7777 \`

Otherwise it's listening on localhost and cannot be reverse proxied and throws 502 bad gateway.

I"m wondering if this may cause other issues with simplelogin?

**Comment by JackyShao01** [comment]
Hi @whisperdancer , I am trying to self host SimpleLogin. Are your proplems solved? Thanks.
